File: app/src/main/java/com/vibhorsrv/cameraids/api/CameraIdentifierAbstract.java
package com.vibhorsrv.cameraids.api;

import com.vibhorsrv.cameraids.model.CameraModel;

/**
 * Base class for camera identifier mechanism
 * <p>
 * Created by Vibhor Srivastava
 *
 * @param <T> Iterable implementation for storing CameraModel objects
 */
public abstract class CameraIdentifierAbstract<T extends Iterable<? super CameraModel>> implements CameraIDs.Identifier<T> {
    protected T cameraModels;

    public CameraIdentifierAbstract(T cameraModels) {
        this.cameraModels = cameraModels;
    }

}

File: app/src/main/java/com/vibhorsrv/cameraids/api/CameraIdentifierAbstract.kt
package com.vibhorsrv.cameraids.api

import com.vibhorsrv.cameraids.model.CameraModel

/**
 * Base class for camera identifier mechanism
 *
 *
 * Created by Vibhor Srivastava
 *
 * @param <T> Iterable implementation for storing CameraModel objects
</T> */
abstract class CameraIdentifierAbstract<T : Iterable<CameraModel>>(
    protected val cameraModels: T
) : CameraIDs.Identifier<T>
